ext/nnstreamer/tensor_decoder/tensordec-directvideo.c

```diff
@@ -80,19 +159,66 @@ dv_getOutCaps (void **pdata, const GstTensorsConfig * config)
   caps = gst_caps_from_string (DECODER_DV_VIDEO_CAPS_STR);
 
   /* Direct video uses the first tensor only even if it's multi-tensor */
-  switch (config->info.info[0].dimension[0]) {
-    case 1:
-      format = GST_VIDEO_FORMAT_GRAY8;
-      break;
-    case 3:
-      format = GST_VIDEO_FORMAT_RGB;
-      break;
-    case 4:
-      format = GST_VIDEO_FORMAT_BGRx;
-      break;
-    default:
-      format = GST_VIDEO_FORMAT_UNKNOWN;
-      break;
+  channel = config->info.info[0].dimension[0];
+  if (channel == 1) {
+    switch (ddata->format) {
+      case DIRECT_VIDEO_FORMAT_I420:
+        format = GST_VIDEO_FORMAT_I420;
+        break;
+      case DIRECT_VIDEO_FORMAT_GRAY8:
+        format = GST_VIDEO_FORMAT_GRAY8;
+        break;
+      default:
+        GST_ERROR ("Invalid Video Format. With 1 channel, only these formats are available: I420, GRAY8");
+        return NULL;
+        break;
+    }
+  } else if (channel == 3) {
+    switch (ddata->format) {
+      case DIRECT_VIDEO_FORMAT_RGB:
+        format = GST_VIDEO_FORMAT_RGB;
+        break;
+      case DIRECT_VIDEO_FORMAT_BGR:
+        format = GST_VIDEO_FORMAT_BGR;
+        break;
+      default:
+        GST_ERROR ("Invalid Video Format. With 3 channels, only these formats are available: RGB, BGR");
```

Comment:
if channel is 3, default GST_VIDEO_FORMAT_RGB if no option param is given.

Reply:
Besides, it should be able to distinguish between "unspecified" and "others (error)".

Reply:
I set the default to UNKNOWN, and if it is, the format will be set to the default according to the number of channels.